### PR TITLE
NXP-29097: fix operation Repository.Query when parameter 'searchTerm'…

### PR DESCRIPTION
**Pull request by tmartins**
… is set to an empty string

### CODE REVIEW

File: nuxeo-features/nuxeo-automation/nuxeo-automation-features/src/test/java/org/nuxeo/ecm/automation/core/test/SearchOperationTest.java [hunk 1]
```diff
@@ -171,6 +171,27 @@ public void iCanApplySortParametersWithQuery() throws Exception {
         assertEquals("WS2", list.get(1).getTitle());
     }
 
+    /**
+     * @since 2021.11
+     */
+    @Test
+    public void testQueryWithSearchTerm() throws Exception {
+        OperationContext ctx = new OperationContext(session);
+        Map<String, Object> params = new HashMap<String, Object>();
```
Comment (ataillefer) [inline nuxeo-features/nuxeo-automation/nuxeo-automation-features/src/test/java/org/nuxeo/ecm/automation/core/test/SearchOperationTest.java:180]:
```suggestion
        Map<String, Object> params = new HashMap<>();
```

File: nuxeo-features/nuxeo-automation/nuxeo-automation-features/src/main/java/org/nuxeo/ecm/automation/core/operations/services/query/DocumentPaginatedQuery.java [hunk 2]
```diff
@@ -119,6 +121,15 @@ public DocumentModelList run() throws OperationException {
         Long targetPage = currentPageIndex != null ? currentPageIndex.longValue() : null;
         Long targetPageSize = pageSize != null ? pageSize.longValue() : null;
 
+        // NXP-29097: handle empty "searchTerm" parameters in query (referenced by ? character)
+        // test if the query does not wait for any parameter, then it probably contains the searchTerm set to
+        // "", let's nullify "strParameters" variable
+        if (query.indexOf("?") == -1 && strParameters != null && strParameters.size() == 1) {
+            if (Framework.getService(ConfigurationService.class).isBooleanTrue("org.nuxeo.ignore.empty.searchterm")) {
```
Comment (ataillefer) [inline nuxeo-features/nuxeo-automation/nuxeo-automation-features/src/main/java/org/nuxeo/ecm/automation/core/operations/services/query/DocumentPaginatedQuery.java:128]: Merge the 2 ifs?

File: nuxeo-features/nuxeo-automation/nuxeo-automation-features/src/test/java/org/nuxeo/ecm/automation/core/test/SearchOperationTest.java (hunk 1 above)
Comment (tmartins) [reply nuxeo-features/nuxeo-automation/nuxeo-automation-features/src/test/java/org/nuxeo/ecm/automation/core/test/SearchOperationTest.java:180]: I have a warning in Eclipse when I do this change
```
Multiple markers at this line
	- Type safety: The expression of type HashMap needs unchecked conversion to conform to 
	 Map<String,Object>
	- HashMap is a raw type. References to generic type HashMap<K,V> should be parameterized
```

File: nuxeo-features/nuxeo-automation/nuxeo-automation-features/src/main/java/org/nuxeo/ecm/automation/core/operations/services/query/DocumentPaginatedQuery.java (hunk 2 above)
Comment (tmartins) [reply nuxeo-features/nuxeo-automation/nuxeo-automation-features/src/main/java/org/nuxeo/ecm/automation/core/operations/services/query/DocumentPaginatedQuery.java:128]: in term of lisibility, do you think it's an equivalent? I chose to keep the second "if" to split the part related to the code and the part related to configuration. Or is there a better way to have conditions without 2 ifs but keep this information? Or maybe we don't care about it?

File: nuxeo-features/nuxeo-automation/nuxeo-automation-features/src/test/java/org/nuxeo/ecm/automation/core/test/SearchOperationTest.java (hunk 1 above)
Comment (efge) [reply nuxeo-features/nuxeo-automation/nuxeo-automation-features/src/test/java/org/nuxeo/ecm/automation/core/test/SearchOperationTest.java:180]: You probably tried `new HashMap()` but @ataillefer is suggesting `new HashMap<>()`

**Pull request closed without merging**